onnxruntime/core/providers/stvm/stvm_execution_provider.cc

```diff
@@ -0,0 +1,538 @@
+          if(shape1[i] != shape2[i]) {
+            return false;
+          }
+        }
+      } else {
+        return false;
+      }
+
+      return true;
+    }
+
+  private:
+    tvm::runtime::Module* mod_;
+    InputsInfoMap inputs_info_;
+    bool update_output_shapes_ = false;
+    TVMTensorShapes output_shapes_;
+    std::vector<DLTensor> tensors_outputs_;
+};
+
+StvmExecutionProvider::StvmExecutionProvider(const StvmExecutionProviderInfo& info)
+    : IExecutionProvider{kStvmExecutionProvider},
+      info_{info} {
+  ProcessInfo();
+
+  AllocatorCreationInfo default_memory_info = {[](int) {
+                                                 return std::make_unique<STVMAllocator>();
+                                               },
+                                               0, false};
+  allocator_ = CreateAllocator(default_memory_info);
+  InsertAllocator(allocator_);
+
+  // Get environment variables
+  const Env& env_instance = Env::Default();
+
+  const std::string dump_subgraphs_env = env_instance.GetEnvironmentVar(stvm_env_vars::kDumpSubgraphs);
+  if (!dump_subgraphs_env.empty()) {
+    dump_subgraphs_ = std::stoi(dump_subgraphs_env) != 0;
+  }
+}
+
+StvmExecutionProvider::~StvmExecutionProvider() {}
+
+AllocatorPtr StvmExecutionProvider::GetAllocator(int id, OrtMemType mem_type) const {
+  return allocator_;
+}
+
+std::vector<std::unique_ptr<ComputeCapability>>
+StvmExecutionProvider::GetCapability(const GraphViewer& graph_viewer,
+                                     const std::vector<const KernelRegistry*>& /*kernel_registries*/) const {
+  std::vector<std::unique_ptr<ComputeCapability>> result;
+  if (graph_viewer.IsSubgraph()) {
+    return result;
+  }
+
+  // Construct modelproto from graph
+  Model model(graph_viewer.Name(), true, ModelMetaData(), PathString{}, IOnnxRuntimeOpSchemaRegistryList(), graph_viewer.DomainToVersionMap(), std::vector<ONNX_NAMESPACE::FunctionProto>(), *GetLogger());
+  Graph& graph_build = model.MainGraph();
+
+  for (const auto& node : graph_viewer.Nodes()) {
+    std::vector<NodeArg*> inputs, outputs;
+    for (auto input : node.InputDefs()) {
+      auto& n_input = graph_build.GetOrCreateNodeArg(input->Name(), input->TypeAsProto());
+      inputs.push_back(&n_input);
+    }
+    for (auto output : node.OutputDefs()) {
+      auto& n_output = graph_build.GetOrCreateNodeArg(output->Name(), output->TypeAsProto());
+      outputs.push_back(&n_output);
+    }
+    graph_build.AddNode(node.Name(), node.OpType(), node.Description(), inputs, outputs, &node.GetAttributes(), node.Domain());
+  }
+
+  const auto& init_tensors = graph_viewer.GetAllInitializedTensors();
+  for (const auto& tensor : init_tensors) {
+    graph_build.AddInitializedTensor(*(tensor.second));
+  }
+  ORT_ENFORCE(graph_build.Resolve().IsOK());
+
+  ONNX_NAMESPACE::ModelProto model_proto = model.ToProto();
+  model_proto.set_ir_version(ONNX_NAMESPACE::Version::IR_VERSION);
+  auto status = graph_build.Resolve();
+
```

Comment:
The code below doesn't seem to use the copy created above, at least based on a quick look. Unclear if the above code (creating a copy) is being used or not.

Reply:
See comment above, done.